cupy/core/reduction.pxi

```diff
@@ -114,63 +114,93 @@ cpdef tuple _get_out_shape(
     return tuple([shape[i] for i in out_axis])
 
 
-cpdef tuple _get_permuted_args(
-        list args, tuple axis_permutes, tuple shape, tuple params,
-        Py_ssize_t out_ndim):
+cdef tuple _set_permuted_args(
+        list args, tuple axis_permutes, tuple shape, tuple params):
```

Comment:
I feel a little confused that this function updates `args` internally. Could you add a comment?

Reply:
I added comment.